src/Core/src/Platform/Windows/ShadowExtensions.cs

```diff
@@ -24,28 +23,33 @@ public static async Task<CompositionBrush> GetAlphaMaskAsync(this UIElement elem
 
 			try
 			{
-				//For some reason, using  TextBlock and getting the AlphaMask
-				//generates a shadow with a size more smaller than the control size. 
-				if (element is TextBlock textElement)
+				var visual = ElementCompositionPreview.GetElementVisual(element);
```

Comment:
Hey Team!. There are different changes in the PR fixing performance but also sizing issues. We can simplify the changes just including the modifications in the `ShadowExtensions` class. Using the Border Shape to get the AlphaMask directly without recreate a Bitmap is massive performance boost applying shadows to Borders.

Reply:
For Image, TextBlock, Shapes we get the mask using GetAlphaMask https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.image.getalphamask?view=windows-app-sdk-1.5. In the case of using the Clip property and clipping using a Shape, the Shape is used. This PR include changes to also use the Shape used by the Border in the StrokeShape property. For the rest of the possible Views, we do not have a method to obtain the mask directly. Using CreateSpriteVisual create a SpriteVisual object with a rectangle. This would be valid for most cases but not always. For example, using an Entry (TextBox in WinUI). The default style adds a small corner radius. 
![image](https://github.com/dotnet/maui/assets/6755973/a6e1ba3a-b05b-4c03-b3c1-602269afdc5d)
The shadow should follow the silhouette of the control. It is in this case where we use the Bitmap. We have to review it to optimize as well.